docs/proposal/1-authentication-support/README.md

```diff
@@ -0,0 +1,232 @@
+# kstone-api Supports Authentication
+
+## Motivation
+
+Kstone has become a popular solution for etcd management. Due to lack of authentication ability, it has safety vulnerability and is not recommended using in production environment. To improve safety features of kstone, we'd like to support authentication in kstone.
+
+## Goals
+
+* Designed to support multiple authentication plugins.
+* Strong security (password encryption, token expiration time, password reset only, etc.).
+* Built-in username/password authentication, the default is based on bearer token authentication (jwt).
+ 
+## Non-Goals
+
+* Implement a complex RBAC authentication system.
+* Implement a complete user management system(only create an admin user by default).
+
+## Use Cases
+
+Visiting [kstone-dashboard](https://github.com/tkestack/kstone-dashboard) home page, it redirects to the login page.
```

Comment:
Any request to access kstone-api without authentication will directly return a HTTP 401 Unauthorized
error.